frontend/apps/web/components/onboarding-checklist/WelcomeDialog.tsx

```diff
@@ -0,0 +1,161 @@
+'use client';
+import { Dialog, DialogContent, DialogTitle } from '@/components/ui/dialog';
+import { getErrorMessage } from '@/util/util';
+import {
+  createConnectQueryKey,
+  useMutation,
+  useQuery,
+} from '@connectrpc/connect-query';
+import { GetAccountOnboardingConfigResponse } from '@neosync/sdk';
+import {
+  getAccountOnboardingConfig,
+  setAccountOnboardingConfig,
+} from '@neosync/sdk/connectquery';
+import { DialogDescription } from '@radix-ui/react-dialog';
+import { useQueryClient } from '@tanstack/react-query';
+import { ReactElement, useEffect, useState } from 'react';
+import { toast } from 'sonner';
+import { useAccount } from '../providers/account-provider';
+import { Skeleton } from '../ui/skeleton';
+import Configure from './Configure';
+import Connect from './Connect';
+import StepProgress from './StepProgress';
+import Sync from './Sync';
+import WelcomeOverview from './WelcomeOverview';
+import WelcomeRouter from './WelcomeRouter';
+
+export interface FormStep {
+  name: string;
+  component: JSX.Element;
+}
+
+export default function WelcomeDialog(): ReactElement {
+  const { account } = useAccount();
+  const {
+    data,
+    isLoading,
+    isFetching: isValidating,
+    error,
+  } = useQuery(
+    getAccountOnboardingConfig,
+    { accountId: account?.id ?? '' },
+    { enabled: !!account?.id }
+  );
+  const queryclient = useQueryClient();
+  const { mutateAsync: setOnboardingConfigAsync } = useMutation(
+    setAccountOnboardingConfig
+  );
+
+  const [currentStep, setCurrentStep] = useState<number>(0);
```

Comment:
Was thinking about suggesting that we move away form this numbered system and instead use a narrow string type and transition to different states. That would make this code a lot more readable. So instead of doing "currentStep + 1) it's instead `setStep('welcome')` `setStep('connect')` etc.

So instead of having `FormStep.name` be a wide string, instead make it something like this:

```
type FormStepName = "welcome" | "connect" | "configure" | "execute" | "router";
export interface FormStep {
  name: FormStepName;
  component: JSX.Element;
}
```

Then the currentStep becomes:
```
const [currentStep, setCurrentStep] = useState<FormStepName>('welcome');
```
A lot more readable.